format/leveldb/leveldb_ldb.go

```diff
@@ -0,0 +1,265 @@
+package leveldb
+
+// https://github.com/google/leveldb/blob/main/doc/table_format.md
+// https://github.com/google/leveldb/blob/main/doc/impl.md
+// https://github.com/google/leveldb/blob/main/doc/index.md
+
+import (
+	"bytes"
+	"embed"
+	"hash/crc32"
+
+	"github.com/golang/snappy"
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/pkg/bitio"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/interp"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+//go:embed leveldb_ldb.md
+var leveldbFS embed.FS
+
+func init() {
+	interp.RegisterFormat(
+		format.LDB,
+		&decode.Format{
+			Description: "LevelDB Table",
+			Groups:      []*decode.Group{format.Probe},
+			DecodeFn:    ldbDecode,
+		})
+	interp.RegisterFS(leveldbFS)
```

Comment:
Some formats has a `torepr` implementation which is a function to convert the decode tree inside more user representation, not sure if that is interesting for leveldb? for example bson has https://github.com/wader/fq/blob/master/format/bson/bson.jq also don't forget to register the format-overloaded--function https://github.com/wader/fq/blob/master/format/bson/bson.go#L24 (it's part for a hack to make is possible to have kind of "polymorphic" functions)

Reply:
Thanks for the pointers. It's not clear to me in which scenarios you'd use `torepr`.

Reply:
@wader Ah, I think I get it! This was a helpful example for me. Thank you. :)
Yes, this applies here, especially for descriptors. However, this reminds me of another question I have. I will ask below.